makelib/common.mk

```diff
@@ -93,37 +93,57 @@ REMOTE_NAME ?= origin
 
 # all supported platforms we build for this can be set to other platforms if desired
 # we use the golang os and arch names for convenience
-PLATFORMS ?= darwin_amd64 windows_amd64 linux_amd64 linux_arm64
+PLATFORMS ?= darwin_amd64 darwin_arm64 windows_amd64 linux_amd64 linux_arm64
 
 # Set the host's OS. Only linux and darwin supported for now
 HOSTOS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
 ifeq ($(filter darwin linux,$(HOSTOS)),)
 $(error build only supported on linux and darwin host currently)
 endif
 
-# Set the host's arch. Only amd64 support for now
+# Set the host's arch. Only amd64 and arm64 support for now
 HOSTARCH := $(shell uname -m)
 ifeq ($(HOSTARCH),x86_64)
-HOSTARCH := amd64
+SAFEHOSTARCH := amd64
+TARGETARCH := amd64
 endif
-ifneq ($(HOSTARCH),amd64)
-	$(error build only supported on amd64 host currently)
+
+# Set safe architectures for supported OSes (darwin and linux).
+# Apple Silicon binaries are not widely available yet
+ifeq ($(HOSTOS),darwin)
+SAFEHOSTARCH := amd64
+TARGETARCH := $(HOSTARCH)
+endif
+
+# If SAFEHOSTARCH and TARGETARCH have not been defined yet, use HOST
+ifeq ($(origin SAFEHOSTARCH),undefined)
+SAFEHOSTARCH := $(HOSTARCH)
+endif
+ifeq ($(origin TARGETARCH), undefined)
+TARGETARCH := $(HOSTARCH)
 endif
+
+ifeq ($(filter amd64 arm64 ,$(TARGETARCH)),)
+	$(error build only supported on amd64 and arm64 host currently)
```

Comment:
when running `make -j4 build` on my intel mac, I get the following error:

```
> make -j4 build
build/makelib/common.mk:127: *** commands commence before first target.  Stop.
```

According too https://stackoverflow.com/a/4713737, i think the indentation of this line is a problem:
> Second, I think it is choking on the indentation of the $(error ...) expression. While the expression resolves to an empty string, there is still a tab character at the start of the line, which indicates a command, which in turn cannot exist outside a rule.

> I think using spaces rather than tabs to indent would work.

After removing this indentation, to match more what we see on [line 101](https://github.com/upbound/build/pull/139/files#diff-c5171823475b9e22b81c11c127bc77010a66782aad45163f942bfef1decef0bdR101), I get this error:

```
> make -j4 build
build/makelib/common.mk:127: *** build only supported on amd64 and arm64 host currently.  Stop.
```

Adding some debug printing, it looks like my `TARGETARCH` is `x86_64`. It was probably set that way on [line 115](https://github.com/upbound/build/pull/139/files#diff-c5171823475b9e22b81c11c127bc77010a66782aad45163f942bfef1decef0bdR115)